gluten-core/src/main/scala/io/glutenproject/utils/FallbackUtil.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.glutenproject.utils
+
+import io.glutenproject.execution.{GlutenColumnarToRowExecBase, GlutenRowToColumnarExec}
+import io.glutenproject.extension.GlutenPlan
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.execution.{InputAdapter, SparkPlan, WholeStageCodegenExec}
+import org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec
+import org.apache.spark.sql.execution.exchange.BroadcastExchangeExec
+
+/** attention: if AQE is enable,This method will only be executed correctly after the execution plan
+ *  is fully determined
+ */
+
+object FallbackUtil extends Logging {
+
+  def skip(plan: SparkPlan): Boolean = {
+    var skip = false
+    plan match {
+      case _: GlutenColumnarToRowExecBase =>
+        skip = true
+      case _: GlutenRowToColumnarExec =>
+        skip = true
+      case WholeStageCodegenExec(_) =>
+        skip = true
+      case InputAdapter(_) =>
+        skip = true
+      case BroadcastExchangeExec(_, _) =>
+        skip = true
+      case AdaptiveSparkPlanExec(_, _, _, _, _) =>
+        skip = true
+      case _ =>
+    }
+    return skip
+  }
+
+  def isFallback(plan: SparkPlan): Boolean = {
+    val fallbackOperator = plan.collect {
+      case plan if !plan.isInstanceOf[GlutenPlan] && !skip(plan) =>
+        plan
+    }
+    if (!fallbackOperator.isEmpty) {
+      log.info(s"gluten fallback operator: {$fallbackOperator}")
```

Comment:
Can `fallbackOperator` be printable?

Reply:
Looks not an issue here. Please ignore the comment.

Reply:
I changed the printing method here to make it better for each operator to occupy one line